packages/mosaic/sql/src/functions/list.ts

```diff
@@ -0,0 +1,58 @@
+import type {ExprValue, MaybeArray} from '../types.js';
+import { asLiteral } from '../util/ast.js';
+import { argsList, fn } from '../util/function.js';
+import { ListNode } from "../ast/list.js";
+import {ExprNode} from "../ast/node.js";
+
+function listFn(name: string, expr: MaybeArray<ExprValue>, ...args: unknown[]) {
+  return fn(name, expr, ...(argsList(args).map(asLiteral)));
+}
+
+/**
+ * Return a SQL AST node for a literal value. The supported types are
+ * null, string, number, boolean, Date, and RegExp. Otherwise, the
+ * input value will be directly coerced to a string.
+ * @param values
+ */
+export function list(values: ExprNode[]) {
```

Comment:
Maybe makes sense to instead have an `ExprValue[]` type, then run a `.map(asLiteral)` over it? That will allow passing in JS values (string, integer, Date) directly. (`asLiteral` treats strings as literal values, `asNode` is similar but treats strings as column references, both will pass through existing `ExprNode` instances unchanged.)